From 8995d98aefc0a1a1125270039097253ff7fba75f Mon Sep 17 00:00:00 2001
From: Sean Whitton <spwhitton@spwhitton.name>
Date: Sat, 25 Nov 2017 16:56:14 -0700
Subject: [PATCH 2/4] dgit-maint-merge(7): Configure --merge-mode for
 gbp-import-orig(1)

For 3.0 (quilt) source packages, gbp-import-orig(1) defaults to
--merge-mode=replace, which does the wrong thing for patches-applied
repositories (see gbp-import-orig(1)).

Note that we do override this when importing the first upstream
version.  This is needed because git-merge(1) will refuse to merge
unrelated histories.

Thanks to Johannes Schauer for reporting the problem.

Signed-off-by: Sean Whitton <spwhitton@spwhitton.name>
---
 dgit-maint-merge.7.pod | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/dgit-maint-merge.7.pod b/dgit-maint-merge.7.pod
index 995fe8b..fe48609 100644
--- a/dgit-maint-merge.7.pod
+++ b/dgit-maint-merge.7.pod
@@ -146,6 +146,9 @@ Now create I<debian/gbp.conf>:
     pristine-tar = False
     pristine-tar-commit = False
 
+    [import-orig]
+    merge-mode = merge
+
 =back
 
 gbp-import-orig(1) requires a pre-existing upstream branch:
-- 
2.15.0

